python_modules/libraries/dagster-airlift/kitchen-sink/kitchen_sink_tests/integration_tests/test_api.py

```diff
@@ -51,3 +54,28 @@ def test_disable_source_code_retrieval_at_scale(airflow_instance: None) -> None:
     for assets_def in defs.assets:
         metadata = next(iter(cast(AssetsDefinition, assets_def).specs)).metadata
         assert SOURCE_CODE_METADATA_KEY in metadata
+
+
+def test_sensor_iteration_multiple_batches(airflow_instance: None) -> None:
+    """Test that sensor iteration correctly retrieves all sensors when over batch limit."""
+    af_instance = AirflowInstance(
+        auth_backend=AirflowBasicAuthBackend(
+            webserver_url=AIRFLOW_BASE_URL, username=USERNAME, password=PASSWORD
+        ),
+        name=AIRFLOW_INSTANCE_NAME,
+        # Set low list limit, force batched retrieval.
+        batch_dag_runs_limit=1,
+    )
+    defs = build_defs_from_airflow_instance(airflow_instance=af_instance)
+    # create a bunch of runs
+    for i in range(2):
+        run_id = af_instance.trigger_dag("simple_unproxied_dag")
+        af_instance.wait_for_run_completion(dag_id="simple_unproxied_dag", run_id=run_id)
+
+    assert defs.sensors
+    sensor_def = next(iter(defs.sensors))
+    with instance_for_test() as instance:
```

Comment:
might be worth sanity checking that we fetch in multiple passes (e.g. that the `batch_dag_runs_limit` actually applies)